Lower per-process websocket connection limits to match other browser behavior

brentfulgham opened the pull request:
#### 80f4f4ee7deb62df8465a1678d21346fab7c3cb8
<pre>
Lower per-process websocket connection limits to match other browser behavior
<a href="https://bugs.webkit.org/show_bug.cgi?id=247100">https://bugs.webkit.org/show_bug.cgi?id=247100</a>
&lt;rdar://90340929&gt;

Reviewed by Chris Dumez.

Chrome and Firefox place an upper bound on the number of connections that a single web page may have open at the same time.

We should follow this behavior to avoid a poorly written web page impacting system performance. We will use Chrome&apos;s larger
limit to reduce the likelihood of web compatibility issues.

Tested by a new test: http/tests/websocket/tests/hybi/multiple-connections-limit.html

* LayoutTests/http/tests/websocket/tests/hybi/multiple-connections-limit-expected.txt: Added.
* LayoutTests/http/tests/websocket/tests/hybi/multiple-connections-limit.html: Added.
* Source/WebKit/WebProcess/Network/WebSocketChannel.cpp:
(WebKit::WebSocketChannel::connect): Check open connections and fail the connection attempt if we exceed them.
* Source/WebKit/WebProcess/Network/WebSocketChannelManager.h:
(WebKit::WebSocketChannelManager::hasReachedSocketLimit const): Added.

Canonical link: <a href="https://commits.webkit.org/258488@main">https://commits.webkit.org/258488@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c8d3d40e424e7590a26d00781b818f9210689f0e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/102050 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/11193 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/35117 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/111374 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/171563 "Built successfully and passed tests") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/12165 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/2104 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/94439 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/109117 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/107831 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/9310 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/92578 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/37136 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/91190 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/24060 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/78860 "Found 2 new API test failures: /WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/accessible/event-listener, /WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/accessible/state-changed (failure)") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/4757 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/25488 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/4862 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/1936 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/10924 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/44978 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5821 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/6611 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebKit/WebProcess/Network/WebSocketChannelManager.h:
```diff
@@ -41,6 +41,8 @@ namespace WebKit {
 
 class WebSocketChannelManager {
 public:
+    static constexpr size_t maximumSocketCount = 200;
```
We should write a comment to explain why we chose 200.

cdumez reviewed (approved):
LGTM

brentfulgham committed:
Lower per-process websocket connection limits to match other browser behavior
https://bugs.webkit.org/show_bug.cgi?id=247100
<rdar://90340929>

Reviewed by Chris Dumez.

Chrome and Firefox place an upper bound on the number of connections that a single web page may have open at the same time.

We should follow this behavior to avoid a poorly written web page impacting system performance. We will use Chrome's larger
limit to reduce the likelihood of web compatibility issues.

Tested by a new test: http/tests/websocket/tests/hybi/multiple-connections-limit.html

* LayoutTests/http/tests/websocket/tests/hybi/multiple-connections-limit-expected.txt: Added.
* LayoutTests/http/tests/websocket/tests/hybi/multiple-connections-limit.html: Added.
* Source/WebKit/WebProcess/Network/WebSocketChannel.cpp:
(WebKit::WebSocketChannel::connect): Check open connections and fail the connection attempt if we exceed them.
* Source/WebKit/WebProcess/Network/WebSocketChannelManager.h:
(WebKit::WebSocketChannelManager::hasReachedSocketLimit const): Added.

Canonical link: https://commits.webkit.org/258488@main
